docarray/document/mixins/multimodal.py

```diff
@@ -97,34 +97,61 @@ def _from_dataclass(cls, obj) -> 'Document':
 
         return root
 
-    def get_multi_modal_attribute(self, attribute: str) -> 'DocumentArray':
-        from docarray import DocumentArray
-
+    def _get_mm_attr_postion(self, attr):
         if not self.is_multimodal:
             raise ValueError(
                 'the Document does not correspond to a Multi Modal Document'
             )
 
-        if attribute not in self._metadata['multi_modal_schema']:
+        if attr not in self._metadata['multi_modal_schema']:
+            raise ValueError(f'the Document schema does not contain attribute {attr}')
+
+        pos = self._metadata['multi_modal_schema'][attr].get('position')
+        if pos is None:
             raise ValueError(
-                f'the Document schema does not contain attribute {attribute}'
+                f'attribute {attr} is not a valid multi modal attribute.'
+                f' One possible cause is the usage of a non-supported type in the dataclass definition.'
             )
+        return int(pos)
```

Comment:
when would this not be an `int`?

Reply:
Not sure to be honest, in the original implementation this was done (but in a slightly different place), so I kept it.